equipment-requests/src/Code.js

```diff
@@ -0,0 +1,198 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var REQUEST_NOTIFICATION_EMAIL = 'it-requests@example.com';
+
+// Form field titles, used for creating the form and as keys when handling
+// responses.
+/**
+ * Add custom menu items when opening the sheet.
+ */
+function onOpen() {
+  var menu = SpreadsheetApp.getUi().createMenu('Equipment requests')
+      .addItem('Set up', 'setup_')
+      .addItem('Clean up', 'cleanup_')
+      .addToUi();
+}
+
+/**
+ * Set up the form and triggers for the workflow.
+ */
+function setup_() {
+  var ss = SpreadsheetApp.getActiveSpreadsheet();
+  if (ss.getFormUrl()) {
+    var msg = 'Form already exists. Unlink the form and try again.';
+    SpreadsheetApp.getUi().alert(msg);
+    return;
+  }
+  var form = FormApp.create('Equipment Requests')
+      .setCollectEmail(true)
+      .setDestination(FormApp.DestinationType.SPREADSHEET, ss.getId())
+      .setLimitOneResponsePerUser(false);
+  var availableLaptops = [
+    '15" high Performance Laptop (OS X)',
+    '15" high Performance Laptop (Windows)',
+    '15" high performance Laptop (Linux)',
+    '13" lightweight laptop (Windows)',
+  ];
+  var availableDesktops = [
+    'Standard workstation (Windows)',
+    'Standard workstation (Linux)',
+    'High performance workstation (Windows)',
+    'High performance workstation (Linux)',
+    'Mac Pro (OS X)',
+  ];
+  var availableMonitors = [
+    'Single 27"',
+    'Single 32"',
+    'Dual 24"',
+  ];
+
+  form.addTextItem().setTitle('Employee name').setRequired(true);
+  form.addTextItem().setTitle('Desk location').setRequired(true);
+  form.addDateItem().setTitle('Due date').setRequired(true);
+  form.addListItem().setTitle('Laptop').setChoiceValues(availableLaptops);
+  form.addListItem().setTitle('Desktop').setChoiceValues(availableDesktops);
+  form.addListItem().setTitle('Monitor').setChoiceValues(availableMonitors);
+
+  var sheets = ss.getSheets().forEach(function(sheet) {
+    if (sheet.getFormUrl() == ss.getFormUrl()) {
+      sheet.hideSheet();
+    }
+  });
+  // Start workflow on each form submit
+  ScriptApp.newTrigger('onFormSubmit_')
+      .forForm(form)
+      .onFormSubmit()
+      .create();
+  // Archive completed items every 5m.
+  ScriptApp.newTrigger('processCompletedItems_')
+      .timeBased()
+      .everyMinutes(5)
+      .create();
+}
+
+/**
+ * Cleans up the project (stop triggers, form submission, etc.)
+ */
+function cleanup_() {
```

Comment:
Side note: this could be a good helper in a template.

Reply:
ack